### PR TITLE
Shorten URL in share modal

### PR DESCRIPTION
**Pull request by davidsingal**
- [x] Refactored shared modal
- [x] Fixed types for Tab component
- [x] Showing shorten URL in shared URL modal
- [x] Load and redirect to the correct URL, given a shorten URL

### CODE REVIEW

File: frontend/cypress/e2e/map.cy.js [hunk 1]
```diff
@@ -64,3 +64,49 @@ describe('Analysis should work for Livelihoods zones layer', () => {
     cy.get('#analysisPanelView').should('be.visible');
   });
 });
+
+describe('Share modal should show shorten URL', () => {
+  const shortenUrl = 'http://localhost:3000/share/cb69ec73745eb70c4a5d';
```
Comment (clementprdhomme) [inline frontend/cypress/e2e/map.cy.js:69]: Maybe you can construct this URL by importing the fixture (`share-url-encode.json`)?

File: frontend/src/pages/share/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+
+import type { AxiosResponse } from 'axios';
+import type { GetServerSideProps } from 'next';
+import type { SharedURLData } from 'types/shared-url';
+
+const SharePage: React.FC = () => null;
+
+export const getServerSideProps: GetServerSideProps = async (ctx) => {
```
Comment (clementprdhomme) [inline frontend/src/pages/share/[id].tsx:9]: - If the ID doesn't exist, could we redirect the user to a 404 page?
- Should you try/catch the request?

**Review by clementprdhomme — COMMENTED** [review]
Hey David,

I found out that when the user navigates the map, we keep sending requests to the API to shorten the URL. Could it be possible to only request a short URL when the user opens the share modal?

File: frontend/cypress/e2e/map.cy.js (hunk 1 above)
Comment (davidsingal) [reply frontend/cypress/e2e/map.cy.js:69]: you are right

File: frontend/cypress/e2e/map.cy.js [hunk 3]
```diff
@@ -68,3 +70,49 @@ describe('Analysis should work for Livelihoods zones layer', () => {
...
+  const shortenUrl = `http://localhost:3000/share/${shareUrlEncodeFixture.attributes.uid}`;
+
+  beforeEach(() => {
+    cy.interceptAllRequests();
+
+    cy.intercept('POST', '/api/share', {
+      fixture: 'share-url-encode.json',
+    }).as('shareEncodeRequest');
+
+    cy.intercept('GET', '/api/share/cb69ec73745eb70c4a5d', {
```
Comment (clementprdhomme) [inline frontend/cypress/e2e/map.cy.js:84]: Shouldn't you import the ID from `shareUrlEncodeFixture.attributes.uid` here as well?

File: frontend/src/pages/share/[id].tsx (hunk 2 above)
Comment (clementprdhomme) [reply frontend/src/pages/share/[id].tsx:9]: I get a 500 when the ID doesn't exist.

When loading http://localhost:3000/share/hello, the default Next.js 500 error page is shown.

**Comment by davidsingal** [comment]
Got it, now it should be fixed

**Pull request merged**